src/main/java/org/sagebionetworks/bridge/services/AdherenceService.java

```diff
@@ -488,4 +494,25 @@ protected WeeklyAdherenceReport deriveWeeklyAdherenceFromStudyAdherenceReport(St
         builder.withStudyStartEventId(study.getStudyStartEventId());
         return func.apply(builder.build());
     }
+    
+    public AdherenceStatistics getAdherenceStatistics(String appId, String studyId, Integer adherenceThreshold) {
+        checkNotNull(appId);
+        checkNotNull(studyId);
+        
+        Study study = studyService.getStudy(appId, studyId, true);
+        
+        scheduleService.getScheduleForStudy(appId, study)
+            .orElseThrow(() -> new EntityNotFoundException(Schedule2.class));
```

Comment:
Elsewhere we report no schedule if there's no schedule...do we want to do the same here?

Reply:
Leave this, it's correct, and filed BRIDGE-3273 to change other adherence APIs to behave in a similar manner.